[dynamo] Hide guard_fail_hook behind a flag to improve cache lookup time (+10% DebertaV2)

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100599
* __->__ #100590


For TorchDynamo eager backend, DebertaV2 speedup improves from 0.77x to 0.87x. 

cc @soumith @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

anijain2305 committed:
[dynamo] Hide guard_fail_hook behind a flag to improve cache lookup time

[ghstack-poisoned]

voznesenskym reviewed (approved):
LGTM, lineage here https://github.com/pytorch/pytorch/pull/98280 (included for posterity, comments, discussions, etc)

anijain2305 committed:
Update on "[dynamo] Hide guard_fail_hook behind a flag to improve cache lookup time (+10% DebertaV2)"



For TorchDynamo eager backend, DebertaV2 speedup improves from 0.77x to 0.87x. 

cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Hide guard_fail_hook behind a flag to improve cache lookup time (+10% DebertaV2)"



For TorchDynamo eager backend, DebertaV2 speedup improves from 0.77x to 0.87x. 

cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Hide guard_fail_hook behind a flag to improve cache lookup time (+10% DebertaV2)"



For TorchDynamo eager backend, DebertaV2 speedup improves from 0.77x to 0.87x. 

cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

anijain2305 committed:
Update on "[dynamo] Hide guard_fail_hook behind a flag to improve cache lookup time (+10% DebertaV2)"



For TorchDynamo eager backend, DebertaV2 speedup improves from 0.77x to 0.87x. 

cc soumith voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx desertfire

[ghstack-poisoned]

wconstab commented on torch/_dynamo/utils.py:
```diff
@@ -1095,6 +1095,14 @@ def __call__(self, gm: torch.fx.GraphModule, example_inputs):
         return gm.forward
 
+    def __enter__(self):
+        self.old_report_guard_failure = config.report_guard_failures
```
nice solution here. thanks!

wconstab reviewed (approved):
lgtm, thanks for finding a good way to auto-enable!

anijain2305 commented:
@pytorchbot merge
